src/styles/main.scss

```diff
@@ -1,3 +1,65 @@
+$data-size: 100px;
+$gap: 1px;
+$month-width-size: 7 * ($data-size + $gap);
+
 body {
   margin: 0;
+  display: flex;
+  justify-content: center;
+  align-content: center;
+}
+
+.calendar {
+  display: flex;
+  gap: $gap;
+  flex-wrap: wrap;
+  width: $month-width-size;
+  padding: 10px;
+
+  @for $value from 28 through 31 {
+    &--month-length--#{$value} :nth-child(n + #{$value + 1}) {
+      display: none;
+    }
+  }
+
+  &__data {
+    display: flex;
+    box-sizing: border-box;
+    width: $data-size;
+    height: $data-size;
+    background-color: #eee;
+    border: 1px solid rgb(0, 0, 0);
+    justify-content: center;
+    align-items: center;
+    cursor: pointer;
+
+    &:hover {
+      transform: translateY(-20px);
+      transition: 0.5s;
```

Comment:
don't add transition on hover, now it works only when you hover on a block, when you move cursor away it doesn't work, also list all styles to which you apply transition